lib/vector-core/src/config/proxy.rs

```diff
@@ -163,9 +163,13 @@ impl ProxyConfig {
                     let mut proxy = Proxy::new(self.interceptor().intercept(proxy_scheme), parsed);
                     if let Ok(authority) = Url::parse(url) {
                         if let Some(password) = authority.password() {
+                            let decoded_user = urlencoding::decode(authority.username())
+                                .expect("username is valid UTF-8 because already URI parsed.");
```

Comment:
I'm not sure I am parsing this expectation right. The error is that the username is valid UTF-8? Why is that an error?

Reply:
Oh this is sometimes a heuristic I've seen, maybe its not universal.
Like the expect message is a way to say "this is why I can use unwrap/expect here".
Because if this happens in the field, its a code bug.

The error that is being unwrapped here is a String::invalidutf8 , which would only happen if the username was invalid utf8, but that cant happen because we already called Url::parse() which would have caught that.

I dont mind updating the expect messages to be more aligned with the actual error if you'd like

Reply:
updated in https://github.com/vectordotdev/vector/pull/20868/commits/956d76af2208ab5d127c832900ed44a2160a9c46